packages/dds/tree/api-report/tree.api.md

```diff
@@ -995,12 +1001,16 @@ export type InsertableFlexNode<TSchema extends FlexTreeNodeSchema> = FlattenKeys
 
 // @public
 export type InsertableObjectFromSchemaRecord<T extends RestrictiveReadonlyRecord<string, ImplicitFieldSchema>> = {
-    readonly [Property in keyof T]: InsertableTreeFieldFromImplicitField<T[Property]>;
+    readonly [Property in keyof T as HasDefault<T[Property]> extends false ? Property : never]: InsertableTreeFieldFromImplicitField<T[Property]>;
+} & {
+    readonly [Property in keyof T as HasDefault<T[Property]> extends true ? Property : never]?: InsertableTreeFieldFromImplicitField<T[Property]>;
```

Comment:
I don't know if it improves anything, but I suspect we can simplify this case since intersecting optional with required is required:
```suggestion
    readonly [Property in keyof T?: InsertableTreeFieldFromImplicitField<T[Property]>;
```
Maybe that breaks IntelliSense or something.

Reply:
Ah, yeah that's fair. I'll keep this thread open and see how modifying this affects things once other unknowns have been settled.

Reply:
Not only does this simplify the type expression, it fixed one of the built-in shadowing issues the original type created... Specifically the constructor-shadowing test case now builds correctly...

Reply:
Unfortunately, it is very disruptive to intellisense, as it ends up displaying both the object with the optional props and the object with the required counterparts (when present). And even for a small schema definition, intellisense ends up abbreviating the type, omitting the required side of things. As a result, everything ends up appearing optional when you look at the type ☹️ So I reverted this change.

Reply:
It's possible this might play nicer with `FlattenKeys` added to the mix. I'll experiment with that and see how the intellsense looks.

Reply:
No, that still ends up looking pretty bad IMO.
![image](https://github.com/microsoft/FluidFramework/assets/54606601/f8fcfa4b-65f5-450f-9c20-a7761c412a5f)